src/components/Personalization/index.js

```diff
@@ -40,26 +41,66 @@ const createCollect = collect => {
     });
 };
 
-const createStore = () => {
+const createStorage = () => {
   const storage = [];
 
-  return (selector, meta) => {
-    storage.push({ selector, meta });
+  return {
+    add: ({ selector, meta }) => {
+      storage.push({ selector, meta });
+    },
+    getAll: () => {
+      return storage;
+    }
   };
 };
 
+// Using node selection vs matches selector, because of :eq()
+const matches = (selector, node) => {
+  const nodes = selectNodesWithEq(selector, node);
```

Comment:
Doesn't `selectNodesWithEq` search `node` and all its children for an element matching the selector? If so, and if I understand correctly, it seems like that might not be what you want to use here since you're just trying to see if the current node (not its children) matches the selector. Right?

Reply:
@Aaronius `selectNodesWitEq` is basically similar to `node.querySelectorAll(selector)`. I was going to create a more optimized version something like `matchesSelectorWithEq`, that would use `node.matches(...)`, but the code was way more complex.

Reply:
It seems like using something like `node.querySelectorAll` would result in incorrect behavior here though, unless I'm just misunderstanding something. For example, let's say you have `A` contains `B` contains `C` contains `D`. Let's say `D` is the element that has a class of `foo` and your personalization rule is targeting the selector `.foo`.

The user clicks on `D`. At this point, the code runs `selectNodesWithEq(".foo", D)`, which would return no results, because `D` doesn't contain an element with `.foo` (`D` _is_ the element matching `.foo`). I think you would want to collect at this point and not move on to inspecting `C`.

Anyway, the code continues and runs `selectNodesWithEq(".foo", C)`. It finds that D is a descendant of `C` and matches `.foo`, so data is collected.

The code continues and runs `selectNodesWithEq(".foo", B)`. It finds that D is a descendant of `B` and matches `.foo`, so data is collected.

The code continues and runs `selectNodesWithEq(".foo", A)`. It finds that D is a descendant of `A` and matches `.foo`, so data is collected.

This seems wrong. It seems like you would want to collect on the first iteration (when inspecting `D`) and not on the subsequent iterations (when inspecting `C`, `B`, and `A`).

There's another problem I believe exists. Let's say you have `X` which contains both `Y` and `Z`. Let's say `Y` has a class of `foo` and `Z` does not. The user clicks element `Z`. In this scenario, data would erroneously be collected, because `selectNodesWithEq(".foo", X)` would return `Y` as a result. But the user never clicked Y so nothing should be collected.

Reply:
@Aaronius I have adjusted the matching logic. The way I have implemented is the following:
1. Using the selector find all the elements
2. Iterate through elements and do a reference check against the clicked element
3. If the reference checks passes collect the metadata

I wish it would have been simpler, but because of `eq()` it is really, really non-trivial.